### PR TITLE
Adjustments for deploying extension on Garden runtime cluster by gardener-operator

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform aws

**What this PR does / why we need it**:
To deploy the provider extension into the Garden runtime cluster by gardener-operator with extension class `garden`, some adjustments are needed in the charts. Missing glue code has been added.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Part of https://github.com/gardener/gardener/issues/9635

**Release note**:

```feature operator
Adjustments for additional deployment of extension and admission controller on Garden runtime cluster by gardener-operator.
```

```breaking operator
The Helm charts for the `application` and `runtime` parts of the gardener-extension-admission-aws admission controller have been separated into standalone charts. These charts now assume a Garden setup with a virtual garden. Both charts must be deployed individually: the `runtime` chart on the Garden runtime cluster, and the `application` chart on the virtual garden. Additionally, the intermediate `global` level in the Helm values has been removed, so you may need to adjust your provided values accordingly.
```

/cc @timuthy @oliver-goetz

### CODE REVIEW

**Review by AndreasBurger — COMMENTED** [review]
lgtm

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
I just noticed in my own PR that with these changes the helm charts are not correct anymore. There are a couple of references to images/image tags which overwrite `global.image.`tags in the pipeline definition ([ref](https://github.com/gardener/gardener-extension-provider-aws/blob/7f20607254dd3c64681516df1de919a440a694c3/.ci/pipeline_definitions#L13-L30)).

**Comment by MartinWeindel** [comment]
> I just noticed in my own PR that with these changes the helm charts are not correct anymore. There are a couple of references to images/image tags which overwrite `global.image.`tags in the pipeline definition ([ref](https://github.com/gardener/gardener-extension-provider-aws/blob/7f20607254dd3c64681516df1de919a440a694c3/.ci/pipeline_definitions#L13-L30)).
> 
> A fix could look like [this](https://github.com/gardener/gardener-extension-provider-gcp/pull/905/commits/f600bf1c6ec0bd840c8f1340fe238eac81d6b1c6).

Thanks! Fixed now.

**Review by oliver-goetz — DISMISSED** [review]
thanks 😄

File: charts/gardener-extension-provider-aws/templates/deployment.yaml [hunk 1]
```diff
@@ -67,8 +71,14 @@ spec:
         - --webhook-config-namespace={{ .Release.Namespace }}
         - --webhook-config-service-port={{ .Values.webhookConfig.servicePort }}
         - --webhook-config-server-port={{ .Values.webhookConfig.serverPort }}
+        {{- if .Values.gardener.runtimeCluster.enabled }}
+        - --disable-controllers=backupentry,bastion,controlplane,infrastructure,worker,healthcheck
```
Comment (timuthy) [inline charts/gardener-extension-provider-aws/templates/deployment.yaml:75]:
```suggestion
        - --controllers=backupbucket,dnsrecord
```

Better use positive list?

**Review by timuthy — CHANGES_REQUESTED** [review]
Thanks for the changes, very nice 👏

File: charts/gardener-extension-admission-aws/charts/application/templates/rbac.yaml [hunk 2]
```diff
@@ -21,6 +21,10 @@ rules:
   - secrets
   verbs:
   - get
+  - list
+  - create
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply charts/gardener-extension-admission-aws/charts/application/templates/rbac.yaml:25]: Good question. I've just tested it without, and it works fine. Not sure why it was added.
Thanks for the catch.

File: charts/gardener-extension-admission-aws/charts/application/templates/rbac.yaml [hunk 3]
```diff
@@ -41,6 +45,30 @@ rules:
...
+  - watch
+  - update
+- apiGroups:
+  - events.k8s.io
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+  - update
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply charts/gardener-extension-admission-aws/charts/application/templates/rbac.yaml:71]: Lease is on the runtime cluster. Thanks for catching. Maybe introduced during some inconsisten intermediate deployment.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**